asimov

```diff
@@ -16,71 +17,78 @@
 # @author  Steve Grunwell
 # @license MIT
 
-readonly FILEPATHS=(
-    "vendor ../composer.json"
-    "node_modules ../package.json"
-    ".vagrant ../Vagrantfile"
-    "bower_components ../bower.json"
-    "target ../pom.xml"
-    ".stack-work ../stack.yaml"
+readonly ASIMOV_ROOT=~
+
+# Paths to unconditionally skip over. This prevents Asimov from modifying the
+# Time Machine exclusions for these paths (and decendents). It has an important
+# side-effect of speeding up the search.
+readonly ASIMOV_SKIP_PATHS=(
+    ~/Library
 )
 
-# Given a directory path, determine if the corresponding file (relative
-# to that directory) is available.
+# A list of "directory"/"sentinel" pairs.
+# Each directory will be excluded from backup iff the sentinel file exists
+# alongside.
 #
-# For example, when looking at a /vendor directory, we may choose to
-# ensure a composer.json file is available.
-dependency_file_exists() {
-    read -r path;
-
-    while read -r path; do
-        safe_filepath_suffix=$(md5 -qs "$(basename "$path")")
-        filename="filepath_${safe_filepath_suffix}"
-        if [ -f "${path}/${!filename}" ]; then
-            echo "$path"
-        fi
-    done
-}
+# For example, 'node_modules package.json' means "exclude node_modules/ from
+# backup if there is a package.json next to it."
+readonly ASIMOV_VENDOR_DIR_SENTINALS=(
+    '.stack-work stack.yaml'
+    '.vagrant Vagrantfile'
+    'bower_components bower.json'
+    'node_modules package.json'
+    'target pom.xml'
+    'vendor composer.json'
+)
 
-# Exclude the given path from Time Machine backups.
+# Exclude the given paths from Time Machine backups.
+# Reads the newline-separated list of paths from stdin.
 exclude_file() {
-    while read -r path; do
-        if tmutil isexcluded "$path" | grep -q '\[Excluded\]'; then
+    local path
+    while IFS=$'\n' read -r path; do
+        if tmutil isexcluded "${path}" | grep -Fq '[Excluded]'; then
             echo "- ${path} is already excluded, skipping."
```

Comment:
would be nice to see here also the size on disk

Reply:
Was preserving existing behaviour in this PR (except for bug fixing). And `du -hs` can be very slow on large directories, which are likely to be the ones you're trying to exclude from backup in the first place.